zerver/actions/message_send.py

```diff
@@ -1681,6 +1705,13 @@ def check_message(
         if realm.mandatory_topics and topic_name == "(no topic)":
             raise JsonableError(_("Topics are required in this organization"))
 
+        topic_creator_user_id = get_topic_creator_user_id(realm.id,stream.recipient_id,topic_name)
+        if stream.is_support_stream and not check_access_based_on_stream_topic_view_policy(sender,stream):
+            topic_creator_user_id = get_topic_creator_user_id(realm.id,stream.recipient_id,topic_name)
+            if topic_creator_user_id and not (sender.id == topic_creator_user_id):
+                raise UnauthorizedTopicAccessError
```

Comment:
So as written, you're calling `topic_creator_user_id` twice when sending a single message; you'll want to avoid that by just calling it once here, and passing it forward if `get_recipient_info` also needs the value.

Reply:
Yeah fixed that.